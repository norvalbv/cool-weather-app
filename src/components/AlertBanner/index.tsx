import React, { ReactElement } from 'react';
import AlertIcon from 'components/SVG/AlertIcon';
import { Alerts } from 'types';
import { convertToDate } from 'utils/convertToTime';

type AlertBannerProps = {
  alerts: Alerts[];
};

const AlertBanner = ({ alerts }: AlertBannerProps): ReactElement => {
  return (
    <>
      {alerts.map((alert, idx) => (
        <div key={`${alert.event}_${idx}`} className="flex items-center gap-4">
          <AlertIcon />
          <div className="flex flex-col gap-2">
            <div>
              <h2 className="font-semibold underline">{alert.event}</h2>
              <span className="inline-block">
                Start Time:&nbsp;{convertToDate(alert.start, false, { type: 'short' })}
              </span>
              <span className="inline-block">
                End Time:&nbsp;{convertToDate(alert.end, false, { type: 'short' })}
              </span>
            </div>
            <p className="text-sm">{alert.description}</p>
          </div>
        </div>
      ))}
    </>
  );
};

export default AlertBanner;
