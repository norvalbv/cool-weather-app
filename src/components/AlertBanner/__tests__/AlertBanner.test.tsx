import React from 'react';
import renderer from 'react-test-renderer';

import AlertBanner from '..';

const alertBannerMockData = {
  sender_name: 'Test Name',
  event: 'Big Snow Storm!',
  start: 123912390,
  end: 12312310239,
  description: 'Uh Oh',
  tags: ['No Tag'],
};

describe('<AlertBanner />', () => {
  test('Should render correctly with mandatory props', () => {
    const tree = renderer
      .create(<AlertBanner alerts={[alertBannerMockData, alertBannerMockData]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
