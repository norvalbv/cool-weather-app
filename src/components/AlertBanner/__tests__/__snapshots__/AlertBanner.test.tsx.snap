// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertBanner /> Should render correctly with mandatory props 1`] = `
[
  <div
    className="flex items-center gap-4"
  >
    <svg>
      AlertIcon
    </svg>
    <div
      className="flex flex-col gap-2"
    >
      <div>
        <h2
          className="font-semibold underline"
        >
          Big Snow Storm!
        </h2>
        <span
          className="inline-block"
        >
          Start Time: 
          01/02/1970
        </span>
        <span
          className="inline-block"
        >
          End Time: 
          05/23/1970
        </span>
      </div>
      <p
        className="text-sm"
      >
        Uh Oh
      </p>
    </div>
  </div>,
  <div
    className="flex items-center gap-4"
  >
    <svg>
      AlertIcon
    </svg>
    <div
      className="flex flex-col gap-2"
    >
      <div>
        <h2
          className="font-semibold underline"
        >
          Big Snow Storm!
        </h2>
        <span
          className="inline-block"
        >
          Start Time: 
          01/02/1970
        </span>
        <span
          className="inline-block"
        >
          End Time: 
          05/23/1970
        </span>
      </div>
      <p
        className="text-sm"
      >
        Uh Oh
      </p>
    </div>
  </div>,
]
`;
